index.html

```diff
@@ -1475,6 +1475,114 @@ <h2>Relationship to Linked Data</h2>
 
       </section>
 
+      <section>
+        <h2>Contexts and Vocabularies</h2>
+
+        <p class="issue" title="(AT RISK) Hash values might change during Candidate Recommendation">
+This section lists cryptographic hash values that might change during the
+Candidate Recommendation phase based on implementer feedback that requires
+the referenced files to be modified.
+        </p>
+        <p>
+Implementations that perform JSON-LD processing MUST treat the following
+JSON-LD context URLs as already resolved, where the resolved document matches
+the corresponding hash values below:
+        </p>
+
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>URL and Media Type</th>
+              <th>Content</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td style="white-space: nowrap;">
+https://www.w3.org/ns/data-integrity/v1<br>
+application/ld+json
+              </td>
+              <td>
+sha256: v/POI0jhSjPansxhJAP1fwepCBZ2HK77fRZfCCyBDs0=<br><br>
+sha3-512: Sg1PLFxKyEYQns9Zr0BoYXtFeDNfrHUDNMkyq4QEWv<wbr>wGIaX1v5xovnCG+dfceZEzr7BhBjm396noZF1HEeCM8g==
+              </td>
+            </tr>
+            <tr>
+              <td style="white-space: nowrap;">
+https://www.w3.org/ns/multikey/v1<br>
+application/ld+json
+              </td>
+              <td>
+sha256: SA+P0RxXl8ZH6xdPcmX71crzeGkCYMPfYjdVT46SF0o=<br><br>
+sha3-512: BgQBN00t6vj1T4uYt5SwdPOhe2BVNY11UEvkVo/rbBv<wbr>fBd+Z09isxhGhf2HkVqMT
+SANah1u7rLk6Uw6GeQGH1w==
+              </td>
+            </tr>
+          </tbody>
+        </table>
+
+        <p>
+Implementations that perform RDF vocabulary processing MUST 1) be implemented
+such that the vocabulary URLs used in the JSON-LD contexts listed above
+resolve to the files listed below, and 2) match the listed cryptographic
+hashes (if specified). The vocabulary content at the URLs listed below are
+normative.
+        </p>
+
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>URL and Media Type</th>
+              <th>Content</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>
+https://w3id.org/security#<br>
+text/html
+              </td>
+              <td>
+https://w3c.github.io/vc-data-integrity/vocab/security/vocabulary.html
```

Comment:
I think the w3id.org URLs are only used as IDs, and when I tried to read the JSON-LD spec, I couldn't find anywhere that the document they resolve to had any normative impact. Did I misread that, and implementers actually have to care about the content? If they don't have to care, you can probably omit the content URL.

Reply:
Yes, correct, the w3id.org URLs are only used as IDs.

The document they resolve to might have more machine-readable information at it that can be used by an RDF reasoner... but that's not a requirement. The only requirement is that it establishes a globally unambiguous identifier.

In the case of vc-data-model and vc-data-integrity, we /do/ put machine-readable information at the end of the IDs, and we also state explicitly what should be at the end of those URLs and that implementations must treat the URL as resolved (per your guidance, @jyasskin), thus ensuring that the identifiers resolve to the machine-readable information that the WG defined.

Reply:
As for w3id.org, it exists as a redirection service (for over 670+ projects). One usage of the service is so that we can establish stable identifiers for vocabulary documents while they're being incubated in W3C CGs, and continue to use those identifiers as the specification goes from experimentation, to incubation, to adoption by an official WG, all the way to a W3C REC. As a specification goes through those processes (and migrates from an organization, to a community group, to a working group, to a TR) we update the w3id.org redirect to the new home/location of the specification over time.

This has helped us not disrupt organizations that are deploying this stuff to production on timelines that are faster than the W3C standards process, which is the case for a variety of production deployments of Verifiable Credentials and Data Integrity.